### PR TITLE
update sildisco module for SimpleSAMLphp 2

### PR DESCRIPTION
**Pull request by briskt**
### Changed
- Re-enabled sildisco for IdP discovery.
- Fixed SP name display on sildisco IdP selection page.
- Use `entityDisplayName` Twig filter for displaying the IdP and SP names.
- Updated test metadata to include `entityid` in every entry, and make `name` an array.

### Removed
- Removed source files copied from SimpleSAMLphp that are no longer used.

### CODE REVIEW

File: modules/material/themes/material/default/selectidp-links.twig [hunk 1]
```diff
@@ -55,31 +55,59 @@
...
-              name="idp_{{ idp.entityid|e('html_attr') }}"
-            >
+            <button class="mdl-button logo-container fill-parent" onclick="setSelectedIdp('{{ idp.entityid|e }}')">
               <div class="image-wrapper">
                 <img
                   class="logo"
-                  id="{{ idp.entityid|e('html_attr') }}"
-                  src="{{ idp.iconurl is defined ? idp.iconurl|e('html_attr') : 'default-logo.png' }}"
+                  id="{{ idp.entityid|e }}"
+                  src="{{ idp.logoURL is empty ? 'default-logo.png' : idp.logoURL }}"
```
Comment (jason-jackson) [inline modules/material/themes/material/default/selectidp-links.twig:69]:
```suggestion
                  src="{{ idp.logoURL|default('default-logo.png') }}"
```

File: modules/material/themes/material/default/selectidp-links.twig [hunk 2]
```diff
@@ -55,31 +55,59 @@
...
                 >
               </div>
+
+              <span class="mdl-color-text--grey-600 logo-caption clickable-caption">
+                {{ idp.logoCaption is empty ? '<br>' : idp.logoCaption|e }}
```
Comment (jason-jackson) [inline modules/material/themes/material/default/selectidp-links.twig:74]:
```suggestion
                {{ idp.logoCaption|e|default('<br>') }}
```

File: modules/material/themes/material/default/selectidp-links.twig [hunk 3]
```diff
@@ -28,10 +28,10 @@
   <header class="mdl-layout__header">
     <div class="mdl-layout__header-row">
       <span class="mdl-layout-title scale-to-parent">
-        {% if spName ?? '' is empty %}
+        {% if sp|entityDisplayName is empty %}
```
Comment (jason-jackson) [inline modules/material/themes/material/default/selectidp-links.twig:31]: Is the pipe correct here and below? Or should it be a `.`?

File: modules/material/themes/material/default/selectidp-links.twig [hunk 4]
```diff
@@ -55,31 +55,59 @@
...
+          title="{{ '{selectidp-links:disabled}'|trans({'%idpName%': idp|entityDisplayName}) }}"
+          onclick="clickedAnyway('{{ idp|entityDisplayName }}')"
+        >
+          <div class="mdl-card__media white-bg fixed-height" layout-children="row" child-spacing="center">
+            <div class="logo-container fill-parent">
+              <div class="image-wrapper">
+                <img
+                  class="logo"
+                  id="{{ idp.entityid|e }}"
+                  src="{{ idp.logoURL is empty ? 'default-logo.png' : idp.logoURL }}"
```
Comment (jason-jackson) [inline modules/material/themes/material/default/selectidp-links.twig:93]: Use default here

File: modules/material/themes/material/default/selectidp-links.twig [hunk 5]
```diff
@@ -55,31 +55,59 @@
...
+                >
+              </div>
+              <span class="mdl-color-text--grey-600 logo-caption">
+                {{ idp.logoCaption is empty ? '<br>' : idp.logoCaption|e }}
```
Comment (jason-jackson) [inline modules/material/themes/material/default/selectidp-links.twig:97]: Use default here

File: modules/sildisco/src/IdPDisco.php [hunk 6]
```diff
@@ -59,13 +51,24 @@ private function getSPEntityIDAndReducedIdpList(): array
...
+        // Creative solution for getting the EntityID from the SPMetadata entry in the state
+        // Source: https://github.com/simplesamlphp/simplesamlphp-module-discopower/blob/5e2e5e9da751104d1553d273cfb2d0bd1e2b57df/src/PowerIdPDisco.php#L231
+        // Before the SimpleSAMLphp 2 upgrade, we added it to the state ourselves by overriding the SAML2.php file
+        parse_str(parse_url($_GET['return'], PHP_URL_QUERY), $returnState);
+        $state = Auth\State::loadState($returnState['AuthID'], 'saml:sp:sso');
+        if ($state && array_key_exists('SPMetadata', $state)) {
+            $spmd = $state['SPMetadata'];
+            $this->log('Updated SP metadata from ' . $this->spEntityId . ' to ' . $spmd['entityid']);
+        }
+        $spEntityId = $spmd['entityid'];
```
Comment (jason-jackson) [inline modules/sildisco/src/IdPDisco.php:63]: Nice, I wonder if this will let us handle some SPs better.

**Review by jason-jackson — APPROVED** [review]
Looks good, but didn't check all the documentation for everything.

File: docker-compose.yml [hunk 7]
```diff
@@ -91,7 +91,10 @@ services:
       - ./development/announcement.php:/data/vendor/simplesamlphp/simplesamlphp/announcement/announcement.php
 
       # Utilize custom metadata
-      - ./development/hub/metadata/:/data/vendor/simplesamlphp/simplesamlphp/metadata/
+      - ./development/hub/metadata/idp-remote.php:/data/vendor/simplesamlphp/simplesamlphp/metadata/idp-remote.php
+      - ./development/hub/metadata/saml20-idp-hosted.php:/data/vendor/simplesamlphp/simplesamlphp/metadata/saml20-idp-hosted.php
+      - ./development/hub/metadata/saml20-sp-hosted.php:/data/vendor/simplesamlphp/simplesamlphp/metadata/saml20-sp-hosted.php
+      - ./development/hub/metadata/sp-remote.php:/data/vendor/simplesamlphp/simplesamlphp/metadata/sp-remote.php
```
[the remark this answers is not in the document]
Comment (briskt) [reply docker-compose.yml:97]: Unfortunately, no. There are two files added in the Dockerfile that disappear if the whole directory is assigned to a volume.

File: modules/material/themes/material/default/selectidp-links.twig (hunk 1 above)
Comment (briskt) [reply modules/material/themes/material/default/selectidp-links.twig:69]: good suggestion

File: modules/material/themes/material/default/selectidp-links.twig (hunk 3 above)
Comment (briskt) [reply modules/material/themes/material/default/selectidp-links.twig:31]: It's correct as a filter. It is a custom filter defined by SimpleSAMLphp.

**Review by jason-jackson — APPROVED** [review]

**Pull request merged**